### PR TITLE
CB-15398: Use the CDP_FREEIPA_REBUILD entitlement

### PR DESCRIPTION
**Pull request by jamisonbennett**
Use the CDP_FREEIPA_REBUILD entitlement for FreeIPA rebuild APIs.

This was tested using unit tests and a local deployment of cloudbreak.

See detailed description in the commit message.

### CODE REVIEW

File: freeipa/src/test/java/com/sequenceiq/freeipa/service/stack/RepairInstancesServiceTest.java [hunk 1]
```diff
@@ -473,6 +473,7 @@ public void testRebuildThrowsWhenSourceStackIsRunning() throws Exception {
 
         when(stackService.getByCrnAndAccountIdEvenIfTerminated(ENVIRONMENT_ID1, ACCOUNT_ID, FREEIPA_CRN)).thenReturn(stack);
         when(stackService.findByEnvironmentCrnAndAccountId(ENVIRONMENT_ID1, ACCOUNT_ID)).thenReturn(Optional.of(stack));
+        when(entitlementService.isFreeIpaRebuildEnabled(eq(ACCOUNT_ID))).thenReturn(true);
```
Comment (hreeve-cloudera) [inline freeipa/src/test/java/com/sequenceiq/freeipa/service/stack/RepairInstancesServiceTest.java:476]: Would it make sense to have a test when this is disabled, to check it throws the expected exception?

**Review by sxxgrc — APPROVED** [review]
Looks good to me, good work.

File: freeipa/src/test/java/com/sequenceiq/freeipa/service/stack/RepairInstancesServiceTest.java (hunk 1 above)
Comment (sxxgrc) [reply freeipa/src/test/java/com/sequenceiq/freeipa/service/stack/RepairInstancesServiceTest.java:476]: Ooh good catch, I second this for completeness.

Comment (jamisonbennett) [reply freeipa/src/test/java/com/sequenceiq/freeipa/service/stack/RepairInstancesServiceTest.java:476]: ok, i added the tests;

**Review by hreeve-cloudera — APPROVED** [review]

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaDescribeService.java [hunk 2]
```diff
@@ -49,6 +54,9 @@ public DescribeFreeIpaResponse describe(String environmentCrn, String accountId)
 
     public List<DescribeFreeIpaResponse> describeAll(String environmentCrn, String accountId) {
         List<Stack> stacks = stackService.findMultipleByEnvironmentCrnAndAccountIdEvenIfTerminatedWithList(environmentCrn, accountId);
+        if (!entitlementService.isFreeIpaRebuildEnabled(accountId)) {
```
Comment (lacikaaa) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaDescribeService.java:57]: Could you move this up one line? No need to fetch anything from the db if there is no entitlement

**Review by lacikaaa — CHANGES_REQUESTED** [review]

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaDescribeService.java (hunk 2 above)
Comment (jamisonbennett) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaDescribeService.java:56]: Good point. Thanks.

**Review by doktoric — APPROVED** [review]

**Pull request merged**